presto-product-tests/src/main/java/io/prestosql/tests/hive/TestHiveStorageFormats.java

```diff
@@ -185,6 +200,37 @@ public void testInsertIntoPartitionedTable(StorageFormat storageFormat)
         query(format("DROP TABLE %s", tableName));
     }
 
+    @Test(dataProvider = "storage_formats_with_null_format", groups = STORAGE_FORMATS)
+    public void testInsertIntoTableWithNullFormat(StorageFormat storageFormat)
+    {
+        String tableName = format("storage_format_%s_with_null_format", storageFormat.getName());
+        String nullFormat = "null_value";
+
+        query(format("CREATE TABLE %s (name VARCHAR) " +
+                        "WITH (format = '%s', null_format = '%s')",
+                tableName,
+                storageFormat.getName(),
+                nullFormat));
+
+        // \N is the default null format
+        String[] values = new String[] {nullFormat, null, "non-null", "", "\\N"};
+
+        Row[] storedValues = Arrays.stream(values).map(Row::row).toArray(Row[]::new);
+        // If you try to put the null format in a table, it will come out as null.
+        storedValues[0] = row((Object) null);
+
+        String placeholders = String.join(", ", Collections.nCopies(values.length, "(?)"));
+
+        query(format("INSERT INTO %s VALUES %s", tableName, placeholders),
```

Comment:
can you also add another test which inserts using hive (or extend this one)?

Reply:
This sounds simple, but I'm running into some unusual issues with it—Hive seems to be insert all of `'null_value'`, `null`, _and_ `'\N'` as null, which is not what I expected, and seems less-correct than the behavior for inserting from Presto. I'm not quite sure what to do about that

Reply:
This is not what I am seeing:

On hive:
```
create table t2 (x string) tblproperties('serialization.null.format'='nullvalue');
insert into table t2 values ('\\N');
insert into table t2 values ('nullvalue');
select * from t2;
+-------+
| t2.x  |
+-------+
| \N    |
| NULL  |
+-------+
```
Is that happening for all the fileformats? Which environment are you running against. I run against HDP3. Are we triggering some Hive bug?

Reply:
It doesn't seem to work when you specify a format that respects `serialization.null.format`.

```
create table t3 (x string) stored as textfile tblproperties('serialization.null.format'='null_value');
insert into t3 values ('null_value'), ('\\N');
select * from t3;
```
```
+-------+--+
| t3.x  |
+-------+--+
| NULL  |
| NULL  |
+-------+--+
```

I'm thinking this isn't a Presto issue, so the solution is to test this a different way (e.g. making it an external table and using a static file, but that's hard except for textfile format, and the integration tests test that).

Reply:
it is textfile by default.

Reply:
I tried to work around it in old Hive but I failed. The only idea I have is to guard this assertion with hive version. So we do not check it for old hive versions which seem buggy.

Reply:
After thinking for a bit, I think the best solution is to just change the test to avoid this issue entirely, so I've just pushed that. It's a bug in Hive, not Presto, and Presto never saw anything wrong, so these tests cover Presto just as well. As an added benefit, the tests are a bit simpler now.